### PR TITLE
Upgrade etcd to 3.5.0

### LINKED ISSUE #102294: Update etcd to v3.5.0
/sig scalability

This issue is tracking process of upgrading etcd to 3.5.0 planned for 1.22 K8s release.

Etcd has recently released Beta versions https://github.com/etcd-io/etcd/issues/12970 and is starting the process of correctness and performance testing. To reduce change of finding performance regression late, we will start qualifying Beta Etcd releases on GKE master branch and will upgrade to 3.5.0 as soon as it is available. 

Steps:
* [x] Do preliminary scale testing of etcd v3.5.0-beta.3 release 
* [x] Release etcd v3.5.0-beta.3 image (https://github.com/kubernetes/kubernetes/pull/102277, https://github.com/kubernetes/k8s.io/pull/2101)
* [x] Release etcd v3.5.0-rc.0 image (https://github.com/kubernetes/kubernetes/pull/102585)
* [x] Upgrade etcd server to v3.5.0-rc.0 (https://github.com/kubernetes/kubernetes/pull/102062)
* [x] Wait for 3.5.0 release and upgrade etcd client (https://github.com/kubernetes/kubernetes/pull/100488)
* [x] Release etcd v3.5.0 image (https://github.com/kubernetes/kubernetes/pull/102909, https://github.com/kubernetes/k8s.io/pull/2236)
* [ ] Upgrade etcd server to v3.5.0

/milestone v1.22

### PR DESCRIPTION
**Pull request by serathius**
/kind cleanup
```release-note
NONE
```
Closes https://github.com/kubernetes/kubernetes/issues/102294

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/approve
For the kubeadm changes.

Thanks.

**Comment by caesarxuchao** [comment]
/assign @jingyih @jpbetz 

Not sure who's the etcd maintainer these days. Could you route this to the proper reviewers? Thank you.

**Comment by serathius** [comment]
ping @bowei @sttts

**Comment by sttts** [comment]
/approve

for sample-apiserver.

**Comment by serathius** [comment]
ping @bowei

**Comment by neolit123** [comment]
we shouldn't release 1.22 with a pinned RC version of etcd.
/priority critical-urgent
/milestone v1.22

File: staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml [hunk 1]
```diff
@@ -26,4 +26,4 @@ spec:
         imagePullPolicy: Never
         args: [ "--etcd-servers=http://localhost:2379" ]
       - name: etcd
-        image: quay.io/coreos/etcd:v3.5.0-rc.0
+        image: quay.io/coreos/etcd:v3.5.0
```
Comment (BenTheElder) [inline staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml:29]: this is not new, but is there a reason we're not using the k8s.gcr.io image here?

Comment (pacoxu) [reply staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml:29]: similar to #91257. We'd better move to `k8s.gcr.io`.

**Pull request merged**